File: 9000/9628.py
import array

n = int(input())
value = list(map(int, input().split()))

dp = array.array('i', (0 for i in range(0,n)))
dp[1] = abs(value[1] - value[0]);
for i in range(2, n):
  dp[i] = min(dp[i - 1] + abs(value[i - 1] - value[i]), dp[i - 2] + abs(value[i] - value[i - 2]))

print(dp[n - 1])

File: 9628.cpp
#include <bits/stdc++.h>

using namespace std;

int main()
{
   int n;
   cin >> n;
   int dp[n + 1], value[n + 1];
   for(int i = 0; i < n; i++){
       cin >> value[i];
   }
   dp[0] = 0;
   dp[1] = abs(value[1] - value[0]);
   for(int i = 2; i < n; i++){
       dp[i] = min(dp[i - 1] + abs(value[i - 1] - value[i]), dp[i - 2] + abs(value[i] - value[i - 2]));
   }
   cout << dp[n - 1];
}
